[DOCS] Document source-related restrictions (#124011)
elastic/elasticsearch: docs/reference/elasticsearch/mapping-reference/mapping-source-field.md

Some field types have additional restrictions. These restrictions are documented in the **synthetic `_source`** section of the field type’s [documentation](/reference/elasticsearch/mapping-reference/field-data-types.md).

> Synthetic source is not compatible with [source-only](docs-content://deploy-manage/tools/snapshot-and-restore/source-only-repository.md) snapshot repositories. A different repository type needs to be selected to store indexes configured with synthetic source.

Reviewer: ```suggestion
Synthetic source is not supported in [source-only](docs-content://deploy-manage/tools/snapshot-and-restore/source-only-repository.md) snapshot repositories.
```

Reviewer: I think "not compatible" (or better yet, "not supported") says it all, so I would delete the second sentence. Or, if you want to keep it, I'd invert it and simplify:

```
To store indexes that use synthetic `_source`, choose a different repository type.
```

Author: Added the latter, to provide guidance on options.

::::

> ::::{warning}
> Disabling the `_source` field is not supported for indexes configured with `logsdb` or `time_series` [index mode](/reference/elasticsearch/index-settings/index-modules.md#index-mode-setting).
> ::::

Reviewer: ```suggestion
::::{note}
You can't disable the `_source` field for indexes with [`index_mode`](/reference/elasticsearch/index-settings/index-modules.md#index-mode-setting) set to `logsdb` or `time_series`.
::::
```

Reviewer: It doesn't look great to have 3 stacked callouts here, but I can tackle that in a separate PR.
